src/System.Linq.Expressions/src/System/Linq/Expressions/Compiler/ILGen.cs

```diff
@@ -1080,26 +1080,16 @@ internal static void EmitArray<T>(this ILGenerator il, IList<T> items)
         }
 
         /// <summary>
-        /// Emits an array of values of count size.  The items are emitted via the callback
-        /// which is provided with the current item index to emit.
+        /// Emits an array of values of count size.
         /// </summary>
-        internal static void EmitArray(this ILGenerator il, Type elementType, int count, Action<int> emit)
+        internal static void EmitArray(this ILGenerator il, Type elementType, int count)
         {
             ContractUtils.RequiresNotNull(elementType, nameof(elementType));
-            ContractUtils.RequiresNotNull(emit, nameof(emit));
+            
```

Comment:
Using issue #11325 to track changes from `ContractUtils` to `Debug` asserts where possible. Will make separate PR for this.